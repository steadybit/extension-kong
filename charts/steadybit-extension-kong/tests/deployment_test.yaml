templates:
  - deployment.yaml
tests:
  - it: manifest should match snapshot without TLS
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with TLS
    set:
      tls:
        server:
          certificate:
            fromSecret: server-cert
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with mutual TLS
    set:
      tls:
        server:
          certificate:
            fromSecret: server-cert
        client:
          certificates:
            fromSecrets:
              - client-cert-a
              - client-cert-a
    asserts:
      - matchSnapshot: {}
