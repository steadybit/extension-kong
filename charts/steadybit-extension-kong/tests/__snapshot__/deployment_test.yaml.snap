manifest should match snapshot using podAnnotations and Labels:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
        steadybit.com/extension: "true"
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-extension-kong
      template:
        metadata:
          annotations:
            oneagent.dynatrace.com/injection: "false"
            some-annotation: some-annotation-value
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: steadybit-extension-kong
            some-label: some-label-value
            steadybit.com/discovery-disabled: "true"
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_NAME
                  value: null
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_ORIGIN
                  value: null
              image: ghcr.io/steadybit/extension-kong:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health/liveness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: extension
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health/readiness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 32Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
              volumeMounts: null
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: steadybit-extension-kong
          volumes: null
manifest should match snapshot with TLS:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
        steadybit.com/extension: "true"
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-extension-kong
      template:
        metadata:
          annotations:
            oneagent.dynatrace.com/injection: "false"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: steadybit-extension-kong
            steadybit.com/discovery-disabled: "true"
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_TLS_SERVER_CERT
                  value: /etc/extension/certificates/server-cert/tls.crt
                - name: STEADYBIT_EXTENSION_TLS_SERVER_KEY
                  value: /etc/extension/certificates/server-cert/tls.key
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_NAME
                  value: null
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_ORIGIN
                  value: null
              image: ghcr.io/steadybit/extension-kong:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health/liveness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: extension
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health/readiness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 32Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
              volumeMounts:
                - mountPath: /etc/extension/certificates/server-cert
                  name: certificate-server-cert
                  readOnly: true
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: steadybit-extension-kong
          volumes:
            - name: certificate-server-cert
              secret:
                optional: false
                secretName: server-cert
manifest should match snapshot with extra env vars:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
        steadybit.com/extension: "true"
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-extension-kong
      template:
        metadata:
          annotations:
            oneagent.dynatrace.com/injection: "false"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: steadybit-extension-kong
            steadybit.com/discovery-disabled: "true"
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_NAME
                  value: null
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_ORIGIN
                  value: null
                - name: FOO
                  value: bar
              envFrom:
                - configMapRef: null
                  name: env-configmap
                - name: env-secrets
                  secretRef: null
              image: ghcr.io/steadybit/extension-kong:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health/liveness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: extension
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health/readiness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 32Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
              volumeMounts: null
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: steadybit-extension-kong
          volumes: null
manifest should match snapshot with extra labels:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
        steadybit.com/extension: "true"
        tags.datadoghq.com/service: steadybit-extension
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-extension-kong
      template:
        metadata:
          annotations:
            oneagent.dynatrace.com/injection: "false"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: steadybit-extension-kong
            steadybit.com/discovery-disabled: "true"
            steadybit.com/extension: "true"
            tags.datadoghq.com/service: steadybit-extension
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_NAME
                  value: null
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_ORIGIN
                  value: null
              image: ghcr.io/steadybit/extension-kong:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health/liveness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: extension
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health/readiness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 32Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
              volumeMounts: null
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: steadybit-extension-kong
          volumes: null
manifest should match snapshot with mutual TLS:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
        steadybit.com/extension: "true"
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-extension-kong
      template:
        metadata:
          annotations:
            oneagent.dynatrace.com/injection: "false"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: steadybit-extension-kong
            steadybit.com/discovery-disabled: "true"
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_TLS_SERVER_CERT
                  value: /etc/extension/certificates/server-cert/tls.crt
                - name: STEADYBIT_EXTENSION_TLS_SERVER_KEY
                  value: /etc/extension/certificates/server-cert/tls.key
                - name: STEADYBIT_EXTENSION_TLS_CLIENT_CAS
                  value: /etc/extension/certificates/client-cert-a/tls.crt,/etc/extension/certificates/client-cert-a/tls.crt
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_NAME
                  value: null
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_ORIGIN
                  value: null
              image: ghcr.io/steadybit/extension-kong:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health/liveness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: extension
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health/readiness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 32Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
              volumeMounts:
                - mountPath: /etc/extension/certificates/client-cert-a
                  name: certificate-client-cert-a
                  readOnly: true
                - mountPath: /etc/extension/certificates/server-cert
                  name: certificate-server-cert
                  readOnly: true
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: steadybit-extension-kong
          volumes:
            - name: certificate-client-cert-a
              secret:
                optional: false
                secretName: client-cert-a
            - name: certificate-server-cert
              secret:
                optional: false
                secretName: server-cert
manifest should match snapshot with mutual TLS using containerPaths:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
        steadybit.com/extension: "true"
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-extension-kong
      template:
        metadata:
          annotations:
            oneagent.dynatrace.com/injection: "false"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: steadybit-extension-kong
            steadybit.com/discovery-disabled: "true"
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_TLS_SERVER_CERT
                  value: /etc/tls/server.crt
                - name: STEADYBIT_EXTENSION_TLS_SERVER_KEY
                  value: /etc/tls/server.key
                - name: STEADYBIT_EXTENSION_TLS_CLIENT_CAS
                  value: /etc/tls/ca.crt,/etc/tls/ca2.crt
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_NAME
                  value: null
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_ORIGIN
                  value: null
              image: ghcr.io/steadybit/extension-kong:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health/liveness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: extension
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health/readiness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 32Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
              volumeMounts: null
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: steadybit-extension-kong
          volumes: null
manifest should match snapshot with podSecurityContext:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
        steadybit.com/extension: "true"
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-extension-kong
      template:
        metadata:
          annotations:
            oneagent.dynatrace.com/injection: "false"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: steadybit-extension-kong
            steadybit.com/discovery-disabled: "true"
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_NAME
                  value: null
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_ORIGIN
                  value: null
              image: ghcr.io/steadybit/extension-kong:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health/liveness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: extension
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health/readiness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 32Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
              volumeMounts: null
          securityContext:
            runAsNonRoot: true
            runAsUser: 2222
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: steadybit-extension-kong
          volumes: null
manifest should match snapshot with priority class:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
        steadybit.com/extension: "true"
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-extension-kong
      template:
        metadata:
          annotations:
            oneagent.dynatrace.com/injection: "false"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: steadybit-extension-kong
            steadybit.com/discovery-disabled: "true"
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_NAME
                  value: null
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_ORIGIN
                  value: null
              image: ghcr.io/steadybit/extension-kong:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health/liveness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: extension
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health/readiness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 32Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
              volumeMounts: null
          priorityClassName: my-priority-class
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: steadybit-extension-kong
          volumes: null
manifest should match snapshot without TLS:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        steadybit.com/discovery-disabled: "true"
        steadybit.com/extension: "true"
      name: RELEASE-NAME-steadybit-extension-kong
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: steadybit-extension-kong
      template:
        metadata:
          annotations:
            oneagent.dynatrace.com/injection: "false"
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: steadybit-extension-kong
            steadybit.com/discovery-disabled: "true"
            steadybit.com/extension: "true"
        spec:
          containers:
            - env:
                - name: STEADYBIT_LOG_LEVEL
                  value: INFO
                - name: STEADYBIT_LOG_FORMAT
                  value: text
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_NAME
                  value: null
                - name: STEADYBIT_EXTENSION_KONG_INSTANCE_0_ORIGIN
                  value: null
              image: ghcr.io/steadybit/extension-kong:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /health/liveness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 5
              name: extension
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health/readiness
                  port: 8085
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 1
              resources:
                limits:
                  cpu: 200m
                  memory: 128Mi
                requests:
                  cpu: 50m
                  memory: 32Mi
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
              volumeMounts: null
          securityContext:
            runAsNonRoot: true
            seccompProfile:
              type: RuntimeDefault
          serviceAccountName: steadybit-extension-kong
          volumes: null
